tools/c7n_azure/c7n_azure/template_utils.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2015-2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Generic arm template resource utilities
+"""
+import json
+import os.path
+
+from azure.mgmt.resource.resources.models import DeploymentMode
+from c7n_azure.session import Session
+
+from c7n.utils import local_session
+
+
+class TemplateUtilities(object):
+    def __init__(self):
+        s = local_session(Session)
+        #: :type: azure.mgmt.resource.ResourceManagementClient
+        self.client = s.client('azure.mgmt.resource.ResourceManagementClient')
+
+    def create_resource_group(self, group_name, group_parameters):
+        self.client.resource_groups.create_or_update(group_name, group_parameters)
+
+    def deploy_resource_template(self, group_name, template_file_name, template_parameters=None):
+        arm_template = self.get_json_template(template_file_name)
+        deployment_properties = {
+            'mode': DeploymentMode.incremental,
+            'template': arm_template,
+        }
+
+        if template_parameters:
+            deployment_properties['parameters'] = template_parameters
+
+        deployment_async_op = self.client.deployments.create_or_update(
+            group_name, group_name, deployment_properties)
+        deployment_async_op.wait()
```

Comment:
The reason the wait is here is the same reason Stefan mentioned above that we can't deploy the functions until the resources have been provisioned. So it has to be synchronous.

Reply:
Decided to move the wait to the place calling this method in case there are scenarios where we deploy ARM templates and they DON'T have to be synchronous.